TestFoundation/TestNSXMLDocument.swift

```diff
@@ -22,20 +22,47 @@ import CoreFoundation
 class TestNSXMLDocument : XCTestCase {
 
     var allTests: [(String, () throws -> Void)] {
-        return [
-            ("test_basicCreation", test_basicCreation),
-            ("test_nextPreviousNode", test_nextPreviousNode),
-            ("test_xpath", test_xpath),
-            ("test_elementCreation", test_elementCreation),
-            ("test_elementChildren", test_elementChildren),
-            ("test_stringValue", test_stringValue),
-            ("test_objectValue", test_objectValue),
-            ("test_attributes", test_attributes),
-            ("test_comments", test_comments),
-            ("test_processingInstruction", test_processingInstruction),
-            ("test_parseXMLString", test_parseXMLString),
-            ("test_prefixes", test_prefixes)
-        ]
+        #if os(OSX) || os(iOS)
+            return [
+                ("test_basicCreation", test_basicCreation),
+                ("test_nextPreviousNode", test_nextPreviousNode),
+                ("test_xpath", test_xpath),
+                ("test_elementCreation", test_elementCreation),
+                ("test_elementChildren", test_elementChildren),
+                ("test_stringValue", test_stringValue),
+                ("test_objectValue", test_objectValue),
+                ("test_attributes", test_attributes),
+                ("test_comments", test_comments),
+                ("test_processingInstruction", test_processingInstruction),
+                ("test_parseXMLString", test_parseXMLString),
+                ("test_prefixes", test_prefixes),
+                ("test_validation_success", test_validation_success),
+                ("test_validation_failure", test_validation_failure),
+                ("test_dtd", test_dtd),
+                ("test_documentWithDTD", test_documentWithDTD),
+                ("test_dtd_attributes", test_dtd_attributes)
+            ]
+        #else // On Linux, currently the tests that rely on NSError are segfaulting in swift_dynamicCast
```

Comment:
There is now! https://bugs.swift.org/browse/SR-585
Possibly related to https://bugs.swift.org/browse/SR-429 also.